packaging/installer/kickstart.sh

```diff
@@ -772,15 +772,64 @@ claim() {
   fi
 
   # shellcheck disable=SC2086
-  if ${ROOTCMD} "${NETDATA_CLAIM_PATH}" -token="${NETDATA_CLAIM_TOKEN}" -rooms="${NETDATA_CLAIM_ROOMS}" -url="${NETDATA_CLAIM_URL}" ${NETDATA_CLAIM_EXTRA}; then
-    progress "Successfully claimed node"
-  else
-    warning "Unable to claim node, you must do so manually."
-    if [ -z "${NETDATA_NEW_INSTALL}" ]; then
-      cleanup
-      trap - EXIT
-      exit 1
-    fi
+  ${ROOTCMD} "${NETDATA_CLAIM_PATH}" -token="${NETDATA_CLAIM_TOKEN}" -rooms="${NETDATA_CLAIM_ROOMS}" -url="${NETDATA_CLAIM_URL}" ${NETDATA_CLAIM_EXTRA}
+  case $? in
+    0)
+      progress "Successfully claimed node"
+      return 0
+      ;;
+    1)
+      warning "Unable to claim node due to invalid claiming options. If you are seeing this message, you’ve probably found a bug."
+      ;;
+    2)
+      warning "Unable to claim node due to issues creating the claiming directory or preparing the local claiming key. Make sure you have a working openssl command and that ${INSTALL_PREFIX}/var/lib/netdata/cloud.d exists, then try again."
+      ;;
+    3)
+      warning "Unable to claim node due to missing dependencies. Usually this means that the Netdata Agent was built without support for Netdata Cloud."
+      ;;
+    4)
+      warning "Failed to claim node due to inabiity to connect to ${NETDATA_CLAIM_URL}. Usually this either means that the specified claiming URL is wrong, or that youare having entworking problems."
+      ;;
+    5)
+      progress "Successfully claimed node, but was not able to notify the Netdata agent. You will need to restart the Netdata service on this node before it will show up in the Cloud."
+      return 0
+      ;;
+    8)
+      warning "Failed to claim node due to an invalid agent ID. You can usually resolve this by removing ${INSTALL_PREFIX}/var/lib/netdata/registry/netdata.public.unique.id and restarting the agent, then trying to claim it again."
```

Comment:
Perhaps `and trying to claim it again.’? It sounds more structurally natural to me to have this as a clause in the main sentence instead of making it it’s own sentence.

Reply:
To be fair, it is really difficult sometimes to review your material, because you do write very precise and the clauses are sound in itself. (This is just me telling you in a strange way that you are super good at writing). 

With that in mind, I do come from a more "mechanical" approach to writing and language. A best practice in technical writing is to strongly contain each idea/ step in one unit - in this case a sentence. 

In a normal documentation page, I would instruct the user like this: 

Failed to claim node due to an invalid agent ID. To resolve the issue: 
1. Remove `${INSTALL_PREFIX}/var/lib/netdata/registry/netdata.public.unique.id`.
2. Restart the agent.
3. Try to claim the agent again.

As you can see, this is very structured and maybe a bit "lifeless". But when things go south many people grow impatient and it helps to give them very structured docs to avoid further frustration. 

AFAIK, we do have formatting constraints in the command line, so we need to work with temporal words that we normally would avoid, like "first" and "then". But in the spirit of keeping ideas separate (and also shorter sentences), I would aim at a sentence per action.

Reply:
To clarify a bit, it’s not the highly structured of your proposed change that feels unnatural or out of place to me here, but the fact that it results in a rather short sentence (and, to a lesser extent, the use of a comma to offset an adverb at the beginning of the sentence).

Perhaps something like this instead:

```suggestion
      warning "Failed to claim node due to an invalid agent ID. You can usually resolve this by removing ${INSTALL_PREFIX}/var/lib/netdata/registry/netdata.public.unique.id and restarting the agent. Then try to claim it again using the same options."
```

That still gives the clear, structured, aspect for the final step, but sounds much more natural to me.

Reply:
I like that. :)